modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -849,4 +852,83 @@ public void setPerformBeanValidation(boolean performBeanValidation) {
     public void setUseOptional(boolean useOptional) {
         this.useOptional = useOptional;
     }
+
+    @Override
+    public String toDefaultValue(Schema p) {
+        // we use a custom version of this function to remove the l, d, and f suffixes from integer and double values
+        // these values are later converted from strings to the primitive types using parseInt, parseLong etc so they
+        // should not include type suffix literals
+        // https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10
+        if (ModelUtils.isArraySchema(p)) {
+            final ArraySchema ap = (ArraySchema) p;
+            final String pattern;
+            if (fullJavaUtil) {
+                pattern = "new java.util.ArrayList<%s>()";
+            } else {
+                pattern = "new ArrayList<%s>()";
+            }
+            if (ap.getItems() == null) {
+                return null;
+            }
+
+            String typeDeclaration = getTypeDeclaration(ap.getItems());
+            Object java8obj = additionalProperties.get("java8");
+            if (java8obj != null) {
+                Boolean java8 = Boolean.valueOf(java8obj.toString());
+                if (java8 != null && java8) {
+                    typeDeclaration = "";
+                }
+            }
+
+            return String.format(Locale.ROOT, pattern, typeDeclaration);
+        } else if (ModelUtils.isMapSchema(p)) {
+            final String pattern;
+            if (fullJavaUtil) {
+                pattern = "new java.util.HashMap<%s>()";
+            } else {
+                pattern = "new HashMap<%s>()";
+            }
+            if (ModelUtils.getAdditionalProperties(p) == null) {
+                return null;
+            }
+
+            String typeDeclaration = String.format(Locale.ROOT, "String, %s", getTypeDeclaration(ModelUtils.getAdditionalProperties(p)));
+            Object java8obj = additionalProperties.get("java8");
+            if (java8obj != null) {
+                Boolean java8 = Boolean.valueOf(java8obj.toString());
+                if (java8 != null && java8) {
+                    typeDeclaration = "";
+                }
+            }
+
+            return String.format(Locale.ROOT, pattern, typeDeclaration);
+        } else if (ModelUtils.isIntegerSchema(p)) {
+            if (p.getDefault() != null) {
+                return p.getDefault().toString();
+            }
+            return null;
+        } else if (ModelUtils.isNumberSchema(p)) {
+            if (p.getDefault() != null) {
+                return p.getDefault().toString();
+            }
+            return null;
```

Comment:
We need a case for Float (isFloatSchema) and add "F" at the end of the default value. Please see the error message below for details.

Reply:
For CodegenProperty, I changed the code back to preserve the "f" suffix for this case. If you need me to change our existing code in AbstractJavaCodegen.java from adding "f" to "F" please let me know.